src/uct/ib/base/ib_iface.c

```diff
@@ -397,12 +401,69 @@ ucs_status_t uct_ib_iface_get_device_address(uct_iface_h tl_iface,
 {
     uct_ib_iface_t   *iface   = ucs_derived_of(tl_iface, uct_ib_iface_t);
 
-    uct_ib_iface_address_pack(iface, &iface->gid, uct_ib_iface_port_attr(iface)->lid,
+    uct_ib_iface_address_pack(iface, &iface->gid_info.gid, uct_ib_iface_port_attr(iface)->lid,
                               (void*)dev_addr);
     return UCS_OK;
 }
 
-int uct_ib_iface_is_reachable(const uct_iface_h tl_iface, const uct_device_addr_t *dev_addr,
+static int uct_ib_iface_is_same_roce_version(uct_ib_roce_version_desc_t *local_roce_ver,
+                                             const uct_ib_address_t *remote_ib_addr)
+{
+    int ret, remote_roce_ver;
+
+    ucs_assert(local_roce_ver->ver != UCT_IB_DEVICE_ROCE_ANY);
+
+    if ((remote_ib_addr->flags >> ucs_ilog2(UCT_IB_ADDRESS_FLAG_LAST)) ==
+         UCT_IB_DEVICE_ROCE_V1) {
+        remote_roce_ver = UCT_IB_DEVICE_ROCE_V1;
+    } else if ((remote_ib_addr->flags >> ucs_ilog2(UCT_IB_ADDRESS_FLAG_LAST)) ==
+                UCT_IB_DEVICE_ROCE_V1_5) {
+        remote_roce_ver = UCT_IB_DEVICE_ROCE_V1_5;
+    } else {
+        ucs_assertv((remote_ib_addr->flags >> ucs_ilog2(UCT_IB_ADDRESS_FLAG_LAST)) ==
+                     UCT_IB_DEVICE_ROCE_V2, "%d", remote_ib_addr->flags);
+        remote_roce_ver = UCT_IB_DEVICE_ROCE_V2;
+    }
+
+    ret = (local_roce_ver->ver == remote_roce_ver);
+
+    if (!ret) {
```

Comment:
i think it would be nicer to remove 'ret' var and return 0 or return 1 directly

Reply:
but i want to keep the debug message

Reply:
what about the debug print?

Reply:
keep it inside the if